### DISCUSSION TITLE
Hydration Error

### DISCUSSION THREAD

**Discussion by florianblaschke** [neuefische/questions#656] [Web]
Hi!

My App is running but since I opened up a new page(addRoast) in /pages I get the following error in my console:

<img width="803" alt="Bildschirmfoto 2023-08-01 um 14 24 27" src="https://github.com/neuefische/questions/assets/127996345/015232fc-e3cd-4e6c-9593-8291fd402c2b">

Any suggestions? 

Here you can find the rest of the code: https://github.com/florianblaschke/capstone-project-beanerd/tree/feature/addRoast/

Thank you for your help!

**Comment by Kheirah** [comment]
Hey there 👋🏼 
I can see two minor things, but they're unlikey to be causing the hydration error. You can try to fix them nonetheless and see if it helps. In your `Form` component:
```js
function showRatio(event) {
    let ara = Number(event.target.value); //THIS LINE CHANGED
    let rob = 100 - ara;

    setArabica(ara);
    setRobusta(rob);
  }
```
Your state is of type `number` but any value from inputs is not; it's of type `string`.

What is more, the input type `text` does not have the `min` and `max` properties:
```js
<StyInput type="text" min="1" max="32" id="roaster" name="roaster" />
```
You may safely remove them. If the hydration error persists after these changes, you can try to temporarily disable strict mode in `next.config.js`:
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: false, //THIS LINE CHANGED
  swcMinify: true,
//...
```
If the error disappears when strict mode is disabled, _you can enable strict mode again and can ignore the error._ In that case, the cause is related to Next's development environment which renders everything twice, once on the server and once on the client, which is the cause for the differing class names. On production the error wouldn't even appear.
